service/matching/matching_engine_test.go

```diff
@@ -3927,6 +3927,7 @@ func defaultTestConfig() *Config {
 	config := NewConfig(dynamicconfig.NewNoopCollection())
 	config.LongPollExpirationInterval = dynamicconfig.GetDurationPropertyFnFilteredByTaskQueue(100 * time.Millisecond)
 	config.MaxTaskDeleteBatchSize = dynamicconfig.GetIntPropertyFnFilteredByTaskQueue(1)
+	config.MatchingDropNonRetryableTasks = dynamicconfig.GetBoolPropertyFn(true) // for tests testing matching non-retryable tasks being dropped
```

Comment:
is it possible to enable this only for the test(s) that care about it? We prefer to keep test config as close as possible to prod config.

Reply:
good point - yes, we can have this. The latest commit has the changes.